client/my-sites/themes/theme-showcase.jsx

```diff
@@ -103,6 +118,19 @@ class ThemeShowcase extends Component {
 		showUploadButton: true,
 	};
 
+	static getDerivedStateFromProps( nextProps, prevState ) {
+		// Activate the "All" tab when searching on "Recommended", since the
+		// search might include some results that are not in the recommended
+		// themes (e.g. WP.org themes).
+		if ( nextProps.search && prevState.tabFilter.key === staticFilters.RECOMMENDED.key ) {
+			return {
+				tabFilter: staticFilters.ALL,
+			};
+		}
```

Comment:
I think we could make the code a bit more readable by destructuring the `nextPros` and `prevState`. Feel free to disagree.
```suggestion
  const { search, tabFilter } = nextProps;
  const { key } = prevState.tabFilter;

  // Activate the "All" tab when searching on "Recommended", since the
  // search might include some results that are not in the recommended
  // themes (e.g. WP.org themes).
  if (search && key === staticFilters.RECOMMENDED.key) {
    return {
      tabFilter: staticFilters.ALL,
    };
  }
```

Reply:
I think `tabFilter.key` is less confusing than just `key`.  Agree with the rest of your suggestion.